spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshMultipleNamespacesIT.java

```diff
@@ -0,0 +1,157 @@
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.configuration.watcher;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import io.kubernetes.client.openapi.models.V1ConfigMapBuilder;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1EnvVar;
+import io.kubernetes.client.openapi.models.V1Service;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.k3s.K3sContainer;
+
+import org.springframework.cloud.kubernetes.integration.tests.commons.Commons;
+import org.springframework.cloud.kubernetes.integration.tests.commons.Phase;
+import org.springframework.cloud.kubernetes.integration.tests.commons.native_client.Util;
+
+import static org.awaitility.Awaitility.await;
+
+class ActuatorRefreshMultipleNamespacesIT {
+
+	private static final String SPRING_CLOUD_K8S_CONFIG_WATCHER_APP_NAME = "spring-cloud-kubernetes-configuration-watcher";
+
+	private static final String WIREMOCK_HOST = "localhost";
+
+	private static final String WIREMOCK_PATH = "/";
+
+	private static final int WIREMOCK_PORT = 80;
+
+	private static final String DEFAULT_NAMESPACE = "default";
+
+	private static final String LEFT_NAMESPACE = "left";
+
+	private static final String RIGHT_NAMESPACE = "right";
+
+	private static final K3sContainer K3S = Commons.container();
+
+	private static Util util;
+
+	@BeforeAll
+	static void beforeAll() throws Exception {
+		K3S.start();
+		Commons.validateImage(SPRING_CLOUD_K8S_CONFIG_WATCHER_APP_NAME, K3S);
+		Commons.loadSpringCloudKubernetesImage(SPRING_CLOUD_K8S_CONFIG_WATCHER_APP_NAME, K3S);
+		util = new Util(K3S);
+		util.createNamespace(LEFT_NAMESPACE);
+		util.createNamespace(RIGHT_NAMESPACE);
+		util.setUpClusterWide(DEFAULT_NAMESPACE, Set.of(DEFAULT_NAMESPACE, LEFT_NAMESPACE, RIGHT_NAMESPACE));
+	}
+
+	@AfterAll
+	static void afterAll() throws Exception {
+		util.deleteNamespace(LEFT_NAMESPACE);
+		util.deleteNamespace(RIGHT_NAMESPACE);
+		Commons.cleanUp(SPRING_CLOUD_K8S_CONFIG_WATCHER_APP_NAME, K3S);
+	}
+
+	@BeforeEach
+	void setup() {
+		configWatcher(Phase.CREATE);
+		util.wiremock(DEFAULT_NAMESPACE, "/", Phase.CREATE);
+	}
+
+	@AfterEach
+	void after() {
+		configWatcher(Phase.DELETE);
+		util.wiremock(DEFAULT_NAMESPACE, "/", Phase.DELETE);
+	}
+
+	/**
```

Comment:
this is what I really wanted to have in this PR, an integration test with reload capability if multiple namespaces.

Reply:
somehow this is the universes payback on me, that I did not write such an integration test in the first place, when I wrote this functionality. Lesson learned.